staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/surroundingobject.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaulting
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FocusAccessorFunc encodes a JSON path through the given object, returning a leave,
```

Comment:
I'm having a hard time following the "encodes a JSON path through the given object" comment, maybe because there's no JSON path parameter... maybe an example would help?

Reply:
will check whether it's still used. Will delete it if not.

It's just an accessor for the node that is plugged into the surrounding object below. It was needed to access the pruned node after the surrounding object was pruned.

Reply:
It's used in the default pruner. Will document it better.

Reply:
done